include/s3select.h

```diff
@@ -2368,8 +2438,10 @@ class base_s3object
 	    i->set_last_call();
 	    i->set_skip_non_aggregate(false);//projection column is set to be runnable
 	    projections_resuls.push_value( &(i->eval()) );
-	  }
-	  result_values_to_string(projections_resuls,result);
+      col_ref.push_back(false);
+      last_key.push_back("");
```

Comment:
`col_ref` and `last_key` are actually constant 
here, it is built over and over in an intensive loop.